static/js/components/GcnEventPage.jsx

```diff
@@ -56,6 +61,79 @@ const useStyles = makeStyles((theme) => ({
   },
 }));
 
+const GcnEventSourcesPage = ({ route, sources }) => {
+  const classes = useStyles();
+  const dispatch = useDispatch();
+  const [sourcesRowsPerPage, setSourcesRowsPerPage] = useState(100);
+
+  const handleSourcesTableSorting = (sortData, filterData) => {
+    dispatch(
+      sourcesActions.fetchGcnEventSources(route.dateobs, {
+        ...filterData,
+        pageNumber: 1,
+        numPerPage: sourcesRowsPerPage,
+        sortBy: sortData.name,
+        sortOrder: sortData.direction,
+      })
+    );
+  };
+
+  const handleSourcesTablePagination = (
+    pageNumber,
+    numPerPage,
+    sortData,
+    filterData
+  ) => {
+    setSourcesRowsPerPage(numPerPage);
+    const data = {
+      ...filterData,
+      pageNumber,
+      numPerPage,
+    };
+    if (sortData && Object.keys(sortData).length > 0) {
+      data.sortBy = sortData.name;
+      data.sortOrder = sortData.direction;
+    }
+    dispatch(sourcesActions.fetchGcnEventSources(route.dateobs, data));
+  };
+
+  // eslint-disable-next-line
+  if (sources.sources.length === 0) {
```

Comment:
```suggestion
  if (sources.sources.length === 0) {
```
why is eslint complaining? 
if you change this to 
```javascript
  if (!sources.sources.length) {
```
does it still complain? we'd like to minimize the amount of eslint disabling we're doing

Reply:
⬆️

Reply:
@mcoughlin I'm wondering why you need to disable eslint here. What's the rule that causes it to complain?

Reply:
I get with your suggestion:

  100:24  error  'sources.sources.length' is missing in props validation  react/prop-types
  120:26  error  'sources.sources.length' is missing in props validation  react/prop-types
  276:21  error  'gncEventSources' is not defined                         no-undef